docs/BUILD_OPTIMIZATION.md

```diff
@@ -0,0 +1,228 @@
+# Build Performance Optimization
```

Comment:
Do we need the memory file if we have this doc? What value in having two sources of truth?